docs/development/APPLICATION_DATA.md

```diff
@@ -0,0 +1,148 @@
+# Sending and Receiving Application Data
+
+## Low Level IO
+
+Sending and receiving any data requires reading or writing from the network. s2n-tls offers two ways to handle IO.
+
+### File descriptors
+
+Applications can set file descriptors using s2n_connection_set_fd / s2n_connection_set_read_fd / s2n_connection_set_write_fd.
```

Comment:
Now I'm thinking about it, I wonder if it would be good to talk briefly about where these fds came from. This is technically outside of our library, but it is important context.